roles/kubernetes/common/tasks/main.yaml

```diff
@@ -37,16 +37,38 @@
 
 - name: Enable cgroup v2 # https://rootlesscontaine.rs/getting-started/common/cgroup2/#enabling-cgroup-v2
   block:
-    - name: Adding GRUB_CMDLINE_LINUX to /etc/default/grub file
+    - name: Retrieve contents of /etc/default/grub
+      ansible.builtin.shell: cat /etc/default/grub
+      register: grub_content
+      changed_when: False
+
+    - name: Retrieve existing GRUB_CMDLINE_LINUX options, if any
+      set_fact:
+        grub_cmdline_original: "{{ grub_content.stdout | regex_search('^GRUB_CMDLINE_LINUX=\"([^\"]*)\"$', '\\1', multiline=True) }}"
+
+    - name: Set {{ option }}={{ value }} if option previously disabled
+      set_fact:
+        grub_cmdline_updated: "{{ grub_cmdline_original.0 | default('') | replace(option + '=0', option + '=' + value) }}"
+
+    - name: Adding {{ option }}={{ value }} if option not present
+      set_fact:
+        grub_cmdline_updated: '{{ (grub_cmdline_updated + " " + option + "=" + value) | trim }}'
+      when: 'option not in grub_cmdline_updated'
+
+    - name: Adding updated GRUB_CMDLINE_LINUX to /etc/default/grub file
       ansible.builtin.lineinfile:
         path: /etc/default/grub
         regexp: '^GRUB_CMDLINE_LINUX='
-        line: GRUB_CMDLINE_LINUX="systemd.unified_cgroup_hierarchy=1"
+        line: 'GRUB_CMDLINE_LINUX="{{ grub_cmdline_updated }}"'
```

Comment:
Just as a comment the extra single quotes help with the string issue here correct? Wanted to make sure they were not being added to the line itself.

Reply:
The single quotes aren't added to the line. It's just YAML formatting; I like to put strings in quotes in YAML, although they could probably be removed without causing a behavior change